File: src/types/request.ts
export interface KeyValuePair {
  key: string;
  value: string;
  enabled: boolean;
}

export interface ApiRequest {
  id: string;
  folder_id: string | null;
  collection_id: string;
  name: string;
  protocol: string;
  method: string;
  url: string;
  headers: string; // JSON string of KeyValuePair[]
  query_params: string; // JSON string of KeyValuePair[]
  body_type: string | null;
  body_content: string | null;
  pre_script: string | null;
  post_script: string | null;
  sort_order: number;
  proto_file: string | null;
  grpc_service: string | null;
  grpc_method: string | null;
  created_at: string;
  updated_at: string;
}

export interface ApiResponse {
  status: number;
  status_text: string;
  latency_ms: number;
  size_bytes: number;
  headers: [string, string][];
  body: string;
  body_type: string;
}

File: src-tauri/src/models/request.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct ApiRequest {
    pub id: String,
    pub folder_id: Option<String>,
    pub collection_id: String,
    pub name: String,
    pub protocol: String,
    pub method: String,
    pub url: String,
    pub headers: String, // JSON array string of KeyValuePair
    pub query_params: String, // JSON array string of KeyValuePair
    pub body_type: Option<String>,
    pub body_content: Option<String>,
    pub pre_script: Option<String>,
    pub post_script: Option<String>,
    pub sort_order: i32,
    pub proto_file: Option<String>,
    pub grpc_service: Option<String>,
    pub grpc_method: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KeyValuePair {
    pub key: String,
    pub value: String,
    pub enabled: bool,
}
